test/integration/controller/jobs/pod/pod_controller_test.go

```diff
@@ -0,0 +1,402 @@
+
+			ginkgo.By("checking that workload is created for pod with the queue name")
+			createdWorkload := &kueue.Workload{}
+			gomega.Eventually(func() error {
+				return k8sClient.Get(ctx, wlLookupKey, createdWorkload)
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+			gomega.Expect(createdWorkload.Spec.PodSets).To(gomega.HaveLen(1))
+
+			gomega.Expect(createdWorkload.Spec.QueueName).To(gomega.Equal("test-queue"), "The Workload should have .spec.queueName set")
+
+			ginkgo.By("checking that pod is unsuspended when workload is assigned")
+			clusterQueue := testing.MakeClusterQueue("cluster-queue").
+				ResourceGroup(
+					*testing.MakeFlavorQuotas("default").Resource(corev1.ResourceCPU, "1").Obj(),
+				).Obj()
+			admission := testing.MakeAdmission(clusterQueue.Name).
+				Assignment(corev1.ResourceCPU, "default", "1").
+				AssignmentPodCount(createdWorkload.Spec.PodSets[0].Count).
+				Obj()
+			gomega.Expect(util.SetQuotaReservation(ctx, k8sClient, createdWorkload, admission)).Should(gomega.Succeed())
+			util.SyncAdmittedConditionForWorkloads(ctx, k8sClient, createdWorkload)
+
+			gomega.Eventually(func() bool {
+				if err := k8sClient.Get(ctx, lookupKey, createdPod); err != nil {
+					return false
+				}
+				return !hasKueueSchedulingGate(createdPod)
+			}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+			gomega.Eventually(func() bool {
+				ok, _ := testing.CheckLatestEvent(ctx, k8sClient, "Started", corev1.EventTypeNormal, fmt.Sprintf("Admitted by clusterQueue %v", clusterQueue.Name))
+				return ok
+			}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+			gomega.Expect(len(createdPod.Spec.NodeSelector)).Should(gomega.Equal(1))
+			gomega.Expect(createdPod.Spec.NodeSelector["kubernetes.io/arch"]).Should(gomega.Equal("arm64"))
+			gomega.Consistently(func() bool {
+				if err := k8sClient.Get(ctx, wlLookupKey, createdWorkload); err != nil {
+					return false
+				}
+				return len(createdWorkload.Status.Conditions) == 2
+			}, util.ConsistentDuration, util.Interval).Should(gomega.BeTrue())
+
+			ginkgo.By("checking that pod is stopped when workload is evicted")
+
+			gomega.Expect(
+				workload.UpdateStatus(ctx, k8sClient, createdWorkload, kueue.WorkloadEvicted, metav1.ConditionTrue,
+					kueue.WorkloadEvictedByPreemption, "By test", "evict"),
+			).Should(gomega.Succeed())
+			util.FinishEvictionForWorkloads(ctx, k8sClient, createdWorkload)
+
+			gomega.Eventually(func() bool {
+				if err := k8sClient.Get(ctx, lookupKey, createdPod); err != nil {
+					return false
+				}
+				return !createdPod.DeletionTimestamp.IsZero()
+			}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+
+			gomega.Expect(createdPod.Status.Conditions).Should(gomega.ContainElement(corev1.PodCondition{
+				Type:    "TerminationTarget",
+				Status:  "True",
+				Reason:  "StoppedByKueue",
+				Message: "By test",
+			}))
+		})
+	})
+
+	ginkgo.When("manageJobsWithoutQueueName is enabled", func() {
```

Comment:
Let's remove this entire When.
The logic to accomplish this functionality is the same as any other job.